src/plugins/visualizations/public/wizard/show_new_vis.tsx

```diff
@@ -66,20 +68,23 @@ export function showNewVisModal({
   document.body.appendChild(container);
   const element = (
     <I18nProvider>
-      <NewVisModal
-        isOpen={true}
-        onClose={handleClose}
-        originatingApp={originatingApp}
-        stateTransfer={getEmbeddable().getStateTransfer()}
-        outsideVisualizeApp={outsideVisualizeApp}
-        editorParams={editorParams}
-        visTypesRegistry={getTypes()}
-        addBasePath={getHttp().basePath.prepend}
-        uiSettings={getUISettings()}
-        savedObjects={getSavedObjects()}
-        usageCollection={getUsageCollector()}
-        application={getApplication()}
-      />
+      <Suspense fallback={<EuiLoadingSpinner />}>
```

Comment:
Since we're mounting this in a div that's just appended to `body` I think the loding spinner here wouldn't show up in any reasonable position (you might test that, by adding some `setTimeout` into the import above to delay loading it).

I think we could instead add a global progress bar at the top of the screen (https://elastic.github.io/eui/#/display/progress), like:

```jsx
<EuiPortal>
  <EuiProgress size="xs" position="fixed" />
</EuiPortal>
```

that would be better visible.

Reply:
You are right, will do it!